src/core/date-formatter.ts

```diff
@@ -0,0 +1,443 @@
+        default:
+          return value.toString();
+      }
+    });
+
+    // Month helper - supports name, abbr, and pad formats
+    Handlebars.registerHelper('month', (value: number, options: any) => {
+      const format = options.hash?.format;
+
+      switch (format) {
+        case 'name':
+          return this.getMonthName(value);
+        case 'abbr':
+          return this.getMonthAbbreviation(value);
+        case 'pad':
+          return value.toString().padStart(2, '0');
+        default:
+          return value.toString();
+      }
+    });
+
+    // Weekday helper - supports name and abbr formats
+    Handlebars.registerHelper('weekday', (value: number, options: any) => {
+      const format = options.hash?.format;
+
+      switch (format) {
+        case 'name':
+          return this.getWeekdayName(value);
+        case 'abbr':
+          return this.getWeekdayAbbreviation(value);
+        default:
+          return value.toString();
+      }
+    });
+
+    // Format embedding helper - allows {{dateFmt:name}} syntax
+    // Note: This helper is mainly for documentation. The actual format embedding
+    // is handled by preprocessing in the format() method to avoid circular issues.
+    Handlebars.registerHelper('dateFmt', (formatName: string, _options: any) => {
+      // This should not be called in normal operation due to preprocessing
+      return `[Unresolved: ${formatName}]`;
+    });
+
+    // Mathematical operations helper
+    Handlebars.registerHelper('math', (value: number, options: any) => {
+      const operation = options.hash?.op;
+      const operand = options.hash?.value;
+
+      if (typeof operand !== 'number') {
+        return value;
+      }
+
+      switch (operation) {
+        case 'add':
+          return value + operand;
+        case 'subtract':
+          return value - operand;
+        case 'multiply':
+          return value * operand;
+        case 'divide':
+          return operand !== 0 ? value / operand : value;
+        case 'modulo':
+          return operand !== 0 ? value % operand : value;
+        default:
+          return value;
+      }
+    });
+
+    // Stardate calculation helper for sci-fi calendars
+    Handlebars.registerHelper('stardate', (year: number, options: any) => {
+      const prefix = options.hash?.prefix || '0';
+      const baseYear = options.hash?.baseYear || year;
+      const dayOfYear = options.hash?.dayOfYear || 1;
+      const precision = options.hash?.precision || 1;
+
+      // Calculate stardate: prefix + (year - baseYear) + dayOfYear
+      // Format: XXYYYY.P where XX is era prefix, YYYY is year offset + day, P is precision
+      const yearOffset = year - baseYear;
+      const stardatePrefix = parseInt(prefix) + yearOffset;
+      const paddedDayOfYear = dayOfYear.toString().padStart(3, '0');
+
+      return `${stardatePrefix}${paddedDayOfYear}.${precision > 0 ? '0'.repeat(precision) : ''}`.replace(
+        /\.$/,
+        '.0'
+      );
+    });
```

Comment:
This is fine here for now, but add a TODO comment about extracting it out of the core code since it is specific to a particular system/setting.